Google.PowerShell/BigQuery/BqJob.cs

```diff
@@ -495,8 +651,34 @@ public Apis.Bigquery.v2.Data.Job DoLoad()
         /// </summary>
         public Apis.Bigquery.v2.Data.Job DoExtract()
         {
-            throw new NotImplementedException(
-                "Extract jobs are not implemented yet.  Use Get-BqTabledata instead.");
+            if (ShouldProcess($"{Source.TableId} (EXTRACT)"))
+            {
+                try
+                {
+                    var extractjob = new Apis.Bigquery.v2.Data.Job()
+                    {
+                        Configuration = new JobConfiguration()
+                        {
+                            Extract = new JobConfigurationExtract()
+                            {
+                                SourceTable = Source,
+                                DestinationUris = DestinationUris,
+                                DestinationFormat = (Type == DataFormats.JSON) ? "NEWLINE_DELIMITED_JSON" : Type.ToString(),
```

Comment:
There are a fair amount of string constants here.  I would put them in variables and comment them.